pkgs/by-name/cr/cryptpad/package.nix

```diff
@@ -14,22 +14,53 @@
 }:
 
 let
-  version = "2025.3.0";
+  version = "2025.6.0";
   # nix version of install-onlyoffice.sh
   # a later version could rebuild from sdkjs/web-apps as per
   # https://github.com/cryptpad/onlyoffice-builds/blob/main/build.sh
-  onlyoffice_build =
+  onlyoffice_oldfetch =
     rev: hash:
     fetchFromGitHub {
       inherit rev hash;
       owner = "cryptpad";
       repo = "onlyoffice-builds";
     };
+
+  # New method for v7+ versions that use ZIP releases
+  onlyoffice_newfetch =
+    version: hash:
+    fetchurl {
+      url = "https://github.com/cryptpad/onlyoffice-editor/releases/download/${version}/onlyoffice-editor.zip";
+      inherit hash;
+    };
+
   onlyoffice_install = oo: ''
     oo_dir="$out_cryptpad/www/common/onlyoffice/dist/${oo.subdir}"
-    cp -a "${onlyoffice_build oo.rev oo.hash}/." "$oo_dir"
-    chmod -R +w "$oo_dir"
-    echo "${oo.rev}" > "$oo_dir/.commit"
+    ${
+      if oo ? "version" then
+        ''
+          mkdir -p "$oo_dir"
+          cd "$oo_dir"
+          unzip ${onlyoffice_newfetch oo.version oo.hash}
+          echo "${oo.version}" > "$oo_dir/.version"
+
+          # Clean up help files and dictionaries as per upstream
+          ${lib.optionalString (oo.subdir == "v7") ''
+            rm -rf web-apps/apps/*/main/resources/help
+            rm -rf web-apps/apps/common/main/resources/help/
+            rm -rf dictionaries/
+          ''}
+        ''
+      else
+        ''
+          cp -a "${onlyoffice_oldfetch oo.rev oo.hash}/." "$oo_dir"
+          chmod -R +w "$oo_dir"
+          echo "${oo.rev}" > "$oo_dir/.commit"
+
+          # Clean up help files as per upstream
+          rm -rf "$oo_dir/web-apps/apps/*/main/resources/help"
```

Comment:
Ah, I hadn't noticed but this wildcard is not expanded, the `*` needs to be outside of quotes to work (for example `lib/node_modules/cryptpad/www/common/onlyoffice/dist/v5/web-apps/apps/spreadsheeteditor/main/resources/help/` was not removed)

I've also fixed that in my branch, that saves quite a bit of space (2.9GB -> 2.2GB)